src/daemon.rs

```diff
@@ -514,13 +527,14 @@ impl Daemon {
             .get_or_else(blockhash, || self.load_blocktxids(blockhash))
     }
 
+
     pub fn gettransaction(
         &self,
         txhash: &Txid,
         blockhash: Option<BlockHash>,
     ) -> Result<Transaction> {
         let mut args = json!([txhash.to_hex(), /*verbose=*/ false]);
-        if let Some(blockhash) = blockhash {
+        if let (true, Some(blockhash)) = (self.getindexinfo(vec!["txindex"])?[0].synced, blockhash) {
```

Comment:
I think the logic is inverted here: it should only add the blockhash if it's **not** synced.
There's also one more problem: according to what you said: "RPC command `getindexinfo "txindex"` returns nothing if txindex=0" this will panic if `txindex=0` because you access it unconditionally.

Also it'd be nice to **not** blow up the whole thing if `indexinfo` can't be obtained - just log it and keep degraded performance.

I suggest:

```rust
let txindex_available = self.getindexinfo(vec!["txindex"])
    .unwrap_or_else(|error| {
        error!("failed to get index info: {}, assuming txindex unavailable", error));
        Vec::new()
    })
    .find_map(|info| {
        match info {
            IndexInfo::TxIndex(txindex) => Some(txindex),
        }
    })
    .map(|txindex| txindex.synced)
    .unwrap_or(false);

    if let (false, Some(blockhash) = (txindex_available, blokhash) {
    // ...
```

Note that `synced` could be mapped directly in `find_map` but I find this clearer. Do you agree @romanz ?

Reply:
Aw I made the stupid mistake x).

Yeah indeed I forgot we need to handle the case where the RPC returns nothing, my bad

Reply:
No problem! :)